ci/do_ci.sh

```diff
@@ -8,22 +8,21 @@ set -e
 echo "building using ${NUM_CPUS} CPUs"
 
 function bazel_release_binary_build() {
-  cd "${ENVOY_CONSUMER_SRCDIR}"
   echo "Building..."
-  bazel --batch build ${BAZEL_BUILD_OPTIONS} -c opt @envoy//source/exe:envoy-static.stripped.stamped
+  cd "${ENVOY_BUILD_DIR}"
```

Comment:
Yeah, this is weird. It's even clean in the container immediately before/after the build, but `bazel-out/volatile-status.txt` says dirty. I'll take a look more in a bit, there's got to be a way.

https://github.com/bazelbuild/bazel/issues/2893 is tracking my concerns over the documentation and support-level for Bazel workspace status and C++. I think it's not release quality yet, much like the coverage - there's in theory a way to do it if you hack around enough and spend O(days) figuring it out yourself, but no documentation way of doing it or integration/regression tests.